openforcefield/typing/engines/smirnoff/parameters.py

```diff
@@ -1429,11 +1431,43 @@ def get_parameter(self, parameter_attrs):
 
         Returns
         -------
-        list of ParameterType objects
+        params : list of ParameterType objects
             A list of matching ParameterType objects
-        """
-        # TODO: This is a necessary API point for Lee-Ping's ForceBalance
-        raise NotImplementedError()
+
+        Examples
+        --------
+
+        Create a parameter handler and populate it with some data.
+
+        >>> from simtk import unit
+        >>> handler = BondHandler(skip_version_check=True)
+        >>> handler.add_parameter(
+        ...     {
+        ...         'smirks': '[*:1]-[*:2]',
+        ...         'length': 1*unit.angstrom,
+        ...         'k': 10*unit.kilocalorie_per_mole/unit.angstrom**2,
+        ...     }
+        ... )
+
+        Look up, from this handler, all parameters matching some SMIRKS pattern
+
+        >>> handler.get_parameter({'smirks': '[*:1]-[*:2]'})
+        [<BondType with smirks: [*:1]-[*:2]  length: 1 A  k: 10 kcal/(A**2 mol)  >]
+
+        """
+        params = list()
+        for attr, value in parameter_attrs.items():
+            for param in self.parameters:
+                if param in params:
+                    continue
+                # TODO: Cleaner accessing of cosmetic attributes
+                # See issue #338
+                if param.attribute_is_cosmetic(attr):
+                    attr = '_' + attr
+                if hasattr(param, attr):
+                    if getattr(param, attr) == value:
```

Comment:
This two lines can be reduced to a single check thanks to the three-argument `getattr`. If you agree this adds clarity, you can rewrite it as: 

```python
if getattr(param, attr, None) == value:
    params.append(param)
```

This will work as long as `None` is not an expected value. Otherwise you need to find another sentinel.